packages/sdk-rtl/src/lookerSDKError.ts

```diff
@@ -0,0 +1,185 @@
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+/* eslint-disable @typescript-eslint/ban-ts-comment, @typescript-eslint/no-redeclare */
+
+// Defensive type-level programming to guard against additional
+// parameters being added to Error, as happened in lib.es2022.error.d.ts
+// (which introduced the errorOptions argument)
+type AugmentErrorOptions<
+  ErrorParameters extends unknown[],
+  AdditionalErrorOptions
+> = ErrorParameters extends [(infer Message)?]
+  ? [Message?, AdditionalErrorOptions?]
+  : ErrorParameters extends [(infer Message)?, (infer ErrorOptions)?]
+  ? [Message?, (ErrorOptions & AdditionalErrorOptions)?]
+  : ErrorParameters extends [
+      (infer Message)?,
+      (infer ErrorOptions)?,
+      ...infer Rest
+    ]
+  ? [Message?, (ErrorOptions & AdditionalErrorOptions)?, ...Rest]
+  : never
+
+interface IErrorDetail {
+  field?: string | null
+  code?: string | null
+  message?: string | null
+  documentation_url: string | null
+}
+
+// This specifies SDK custom error options
+interface ILookerSDKErrorOptions {
+  errors?: IErrorDetail[]
+  documentation_url?: string | null
+}
+
+interface ILookerSDKErrorConstructor {
+  new (
+    ...args: AugmentErrorOptions<
+      ConstructorParameters<ErrorConstructor>,
+      ILookerSDKErrorOptions
+    >
+  ): LookerSDKError
+  (
+    ...args: AugmentErrorOptions<
+      Parameters<ErrorConstructor>,
+      ILookerSDKErrorOptions
+    >
+  ): LookerSDKError
+}
+
+// The subclass and function expression's name should match, so that stack traces look clean.
+// We bind it to a local identifier for clarity, and to perform a type assertion.
+export interface LookerSDKError extends Error {
+  errors?: IErrorDetail[]
+  documentation_url?: string | null
+}
+
+export const LookerSDKError: ILookerSDKErrorConstructor =
+  /* #__PURE__*/ (() => {
+    'use strict'
+    const LookerSDKErrorConstructor = function LookerSDKError(
+      this: LookerSDKError | undefined,
+      ...[
+        message,
+        { errors = [], documentation_url = '', ...errorOptions } = {},
```

Comment:
The default values you use in this destructuring will only apply in the case that `errors` and `documentation_url` are `undefined` or not present in the input object. How do you want to handle `null`?

Reply:
I think `errors` should end up as an empty array if it's not in the payload. That's what I've done for the Kotlin SDK, fwiw, and consuming the result becomes much simpler that way. Our API error payloads should always have `documentation_url` but I think that should also default to an empty string if not provided